backend/apps/readings/analysis.py

```diff
@@ -33,6 +34,95 @@ def total_view_time(self):
                 total_view_time += view_time
         return total_view_time
 
+    @property
+    def run_mean_reading_analysis_for_questions(self):
+        """
+        Runs the analysis on the data loaded from the CSV file by looking at the average
+        read time for each question and the context that the question was given in and
+        prints it in a nice readable format.
+        :return: the info wed like to put on js
+        """
+
+        questions = []
+        contexts = []
+        student_data = self.responses[:]
+        for response in student_data:
+            if response.question.text not in questions:
+                questions.append(response.question.text)
+            if response.context.text not in contexts:
+                contexts.append(response.context.text)
+
+
+        mean_reading_time_results_data = []
+
+        for question in questions:
+            for context in contexts:
+                mean_reading_time_results_data.append(self.mean_reading_time_for_a_question(
+                    question, context))
+
+        return mean_reading_time_results_data
+
+    def mean_reading_time_for_a_question(self, question, context):
+        """
+        Given the student response dicts, computes the mean read time for a
+        specific question (given by its keyword) and the context in which it was asked.
+        Returns the question, context, mean read time, and number of people who read.
+        :param question: string, to determine which question was being asked
+        :param context: string, what the reader thought the reading was
+        :return: tuple, in order of the question asked (full question), the context,
+        the mean read time, and the number of people who read it
+        """
+        mean_time = 0
+        number_of_readers = 0
+        question_count = 0
+        reading_time = []
+        total_question_view_time = 0
+        student_data = self.responses[:]
+        for response in student_data:
+            if question != response.question.text or \
+               context != response.context.text:
+                continue
+            if response.get_parsed_views():
+                number_of_readers += 1
+            for view_time in response.get_parsed_views():
+                reading_time.append(view_time)
+
+        if reading_time:
+            self.remove_outliers(reading_time)
+
+        view_time = 0
+        while view_time < len(reading_time):
+            question_count += 1
+            total_question_view_time += reading_time[view_time]
+            view_time += 1
+
+        if reading_time:
+            mean_time = round(total_question_view_time / len(reading_time), 2)
+
+        return [question, context, mean_time, number_of_readers]
+
+    def remove_outliers(self, reading_time):
```

Comment:
This could be useful elsewhere, btw -- maybe make the argument names more generic (i.e., it doesn't actually have to be a list of reading times; this could take any list of numbers) and move it outside the class to be a helper function available to others?

Reply:
And add a test!